src/test/java/de/tum/in/www1/artemis/exam/QuizPoolIntegrationTest.java

```diff
@@ -0,0 +1,209 @@
+        quizGroup0.setName("Encapsulation");
+        QuizGroup quizGroup1 = new QuizGroup();
+        quizGroup1.setName("Inheritance");
+        QuizGroup quizGroup2 = new QuizGroup();
+        quizGroup2.setName("Polymorphism");
+        QuizQuestion mcQuizQuestion0 = database.createMultipleChoiceQuestion();
+        QuizQuestion mcQuizQuestion1 = database.createMultipleChoiceQuestion();
+        QuizQuestion dndQuizQuestion0 = database.createDragAndDropQuestion();
+        QuizQuestion dndQuizQuestion1 = database.createDragAndDropQuestion();
+        QuizQuestion saQuizQuestion0 = database.createShortAnswerQuestion();
+        mcQuizQuestion0.setTitle("MC 0");
+        mcQuizQuestion1.setTitle("MC 1");
+        dndQuizQuestion0.setTitle("DND 0");
+        dndQuizQuestion1.setTitle("DND 1");
+        saQuizQuestion0.setTitle("SA 0");
+        mcQuizQuestion0.setQuizGroup(quizGroup0);
+        mcQuizQuestion1.setQuizGroup(quizGroup0);
+        dndQuizQuestion0.setQuizGroup(quizGroup1);
+        dndQuizQuestion1.setQuizGroup(quizGroup2);
+        quizPool.setQuizGroups(List.of(quizGroup0, quizGroup1, quizGroup2));
+        quizPool.setQuizQuestions(List.of(mcQuizQuestion0, mcQuizQuestion1, dndQuizQuestion0, dndQuizQuestion1, saQuizQuestion0));
+
+        QuizPool responseQuizPool = request.putWithResponseBody("/api/courses/" + course.getId() + "/exams/" + exam.getId() + "/quiz-pools", quizPool, QuizPool.class,
+                HttpStatus.OK, null);
+
+        assertThat(responseQuizPool.getExam().getId()).isEqualTo(exam.getId());
+        assertThat(responseQuizPool.getQuizGroups()).hasSize(quizPool.getQuizGroups().size()).extracting("name").containsExactly(quizPool.getQuizGroups().get(0).getName(),
+                quizPool.getQuizGroups().get(1).getName(), quizPool.getQuizGroups().get(2).getName());
+        assertThat(responseQuizPool.getQuizQuestions()).hasSize(quizPool.getQuizQuestions().size()).extracting("title", "quizGroup.name").containsExactly(
+                tuple(quizPool.getQuizQuestions().get(0).getTitle(), quizGroup0.getName()), tuple(quizPool.getQuizQuestions().get(1).getTitle(), quizGroup0.getName()),
+                tuple(quizPool.getQuizQuestions().get(2).getTitle(), quizGroup1.getName()), tuple(quizPool.getQuizQuestions().get(3).getTitle(), quizGroup2.getName()),
+                tuple(quizPool.getQuizQuestions().get(4).getTitle(), null));
+
+        QuizGroup quizGroup3 = new QuizGroup();
+        quizGroup3.setName("Exception Handling");
+        QuizQuestion saQuizQuestion1 = database.createShortAnswerQuestion();
+        QuizQuestion saQuizQuestion2 = database.createShortAnswerQuestion();
+        QuizQuestion saQuizQuestion3 = database.createShortAnswerQuestion();
+        saQuizQuestion1.setTitle("SA 1");
+        saQuizQuestion2.setTitle("SA 2");
+        saQuizQuestion3.setTitle("SA 3");
+        saQuizQuestion1.setQuizGroup(quizGroup2);
+        saQuizQuestion2.setQuizGroup(quizGroup3);
```

Comment:
The teamscale warning might be addressed by creating a helper method that creates a quiz group with different parameters.

Reply:
updated